src/Data/URI/Host/Gen.purs

```diff
@@ -0,0 +1,15 @@
+module Data.URI.Host.Gen where
+
+import Prelude
+
+import Control.Monad.Gen as Gen
+import Data.String as S
+import Data.URI.Host (Host(..))
+
+genIPv4 :: forall m. Gen.MonadGen m => m Host
+genIPv4 = do
```

Comment:
I mean they actually just correspond to one byte each. I don't think the decimal notation has a formal specification anywhere, but googling around yields that it's preferred to not allow prefixed 0's as they can lead some number parsers to assume the octal system.